x-pack/plugins/ingest_manager/common/types/models/epm.ts

```diff
@@ -136,13 +158,11 @@ export interface Dataset {
   ingest_pipeline: string;
   vars?: VarsEntry[];
   type: string;
+  streams?: RegistryStream[];
   package: string;
 }
 
-export interface VarsEntry {
-  name: string;
-  default: string;
-}
+export type VarsEntry = Record<string, any>;
```

Comment:
Sorry, I missed this in my first pass. `vars` is more complex than a key/value pair. Here's a quick attempt:

```suggestion
export interface VarsEntry {
  name: string;
  description?: string;
  type: string;
  required?: boolean;
  multi?: boolean;
  default?: string | string[];
  os?: {
    [key: string]: {
      default: string | string[];
    }
  }
};
```
Based on:
```
            {
              "default": [
                "/var/log/nginx/access.log*"
              ],
              "description": "Paths to the nginx access log file.",
              "multi": true,
              "name": "paths",
              "os": {
                "darwin": {
                  "default": [
                    "/usr/local/var/log/nginx/access.log*"
                  ]
                },
                "windows": {
                  "default": [
                    "c:/programdata/nginx/logs/*access.log*"
                  ]
                }
              },
              "required": true,
              "type": "text"
            }
```

Reply:
I saw those additional fields and almost added them, but I noticed the registry has no idea (so no guarantees) about them https://github.com/ruflin/package-registry/blob/590dab8ce992c16ce63c3e74d98839b8a986ca7b/util/dataset.go#L20

I just pushed an update to use the suggested shape, though. Let's start with that and iterate later if need be.